src/uct/base/uct_iface.c

```diff
@@ -310,7 +310,6 @@ int uct_base_iface_is_reachable(const uct_iface_h tl_iface,
 int uct_base_ep_is_connected(const uct_ep_h tl_ep,
                              const uct_ep_is_connected_params_t *params)
 {
-    UCT_EP_IS_CONNECTED_CHECK_DEV_IFACE_ADDRS(params);
```

Comment:
since `params->device_addr` and `params->iface_addr` are still passed to `uct_base_iface_is_reachable()`, shall instead make sure that they are actively set (maybe to NULL for TLs where there is no address) by keeping `UCT_EP_IS_CONNECTED_CHECK_DEV_IFACE_ADDRS()` line 313?

Reply:
I cannot know in this function which addresses should be set because it's per TL.
And I don't think check for NULL for unset addresses are required by the UCT API (AFAIU, non-set addresses can br uninitialized pointers).

Reply:
ok but the check is removed, so now supporting transports cannot check anymore if the parameter was actually provided. i guess this is fine for transport that do not have address, but do you think it could be worth keeping the check and expect transports without addr to pass NULL pointers and setting the flag?

Reply:
I think it's problematic. For example look at knem_iface.c line 84:
`.ep_is_connected       = uct_base_ep_is_connected`
This function is called from wireup.c line 1911:
`        return uct_ep_is_connected(ucp_wireup_get_tl_ep(uct_ep), &params);`
The second arg (`params`) is initialized in `ucp_wireup_fill_is_connected_params`, so theoretically we can pass NULL for non-provided addresses.
But UCT API can also be called from MPI and other apps, and unlike UCP we cannot demand that they will pass NULL pointers.

BTW, the checks for the validity of the pointers are done in is_reachable which is called by `uct_base_ep_is_connected`

Reply:
I see. Is there a way to somehow keep the check for supporting transports (the one that mandate flags+params being set)?

Because `uct_base_ep_is_connected()` calls `uct_base_iface_is_reachable()` which potentially falsely adds `UCT_IFACE_IS_REACHABLE_FIELD_{IFACE,DEVICE_ADDR}` and pointers that might not have been present/valid in the first place, if this check is removed.

Reply:
IMO it's ok since we call uct_iface_is_reachable_v2 that also checks the input parameters, per transport

Reply:
ok app or ucp directly calling `uct_iface_is_reachable_v2()` is still ok with per transport checks. but I think there is this case, app (or maybe ucp):
1. `uct_ep_is_connected()` calls function pointer: `uct_base_ep_is_connected()`, with the now removed flags check.
2. `uct_base_ep_is_connected()` now does not check flags, calls `uct_base_iface_is_reachable()` which _unconditionally_ adds flags with pointer, before calling `uct_iface_is_reachable_v2()`, which cannot check flags, since they are always added.

Reply:
I see, so I think the best thing is to avoid using `uct_base_iface_is_reachable` here, and instead call directly to `uct_iface_is_reachable_v2` (and set only addresses which are passed).